### PR TITLE
Event log table report fixes

### PR DESCRIPTION
**Pull request by jgeurten**
Fixed the following broken experience - 

1. An event with >1 correlated signing events would only show 1 signer event in the table
2. Publisher="Unknown" signing events now are filtered out by the Wizard and not shown to the user
3. Previous commit (#118) broke event sorting in the DataGrid, specifically the event/trigger for EventRowClick(). Fixed sorting experience
4. Also added the ability for a user to move through the table using arrow keys. Now hovering over a row or clicking a row will update the fields in the rule creator UI

### CODE REVIEW

File: WDAC-Policy-Wizard/app/src/EventLogRuleConfiguration.cs [hunk 1]
```diff
@@ -142,7 +139,39 @@ private void AddButton_Click(object sender, EventArgs e)
...
         /// <summary>
-        /// Populates the custom rules panel UI with the contents from the CiEvent list
+        /// Populates the custom rules panel with CiEvent on table row highlight event
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void RowSelectionChanged(object sender, EventArgs e)
+        {
+            // Try to cast sender as Datagrid. If successful, the user highlighted (arrow keys or mouse click) a new row
+            var grid = sender as DataGridView;
```
Comment (jgeurten) [inline WDAC-Policy-Wizard/app/src/EventLogRuleConfiguration.cs:149]: @bowerscd-corp hope this makes you happy! Good suggestion, though. Much easier than try-catch casting or smth else

**Review by jsuther1974 — APPROVED** [review]

File: WDAC-Policy-Wizard/app/src/EventLogRuleConfiguration.cs [hunk 2]
```diff
@@ -142,7 +139,39 @@ private void AddButton_Click(object sender, EventArgs e)
...
+            int selectedRow = grid.CurrentRow.Index;
+            if (selectedRow >= this.CiEvents.Count)
+            {
+                return;
+            }
+
+            // Set the UI
+            ResetCustomRulesPanel();
+            SetPublisherPanel(
+                this.CiEvents[selectedRow].SignerInfo.IssuerName,
```
Comment (bowerscd-corp) [inline WDAC-Policy-Wizard/app/src/EventLogRuleConfiguration.cs:164]: This is also a bit weird - I would move the first parameter up one line and align the rest.

Comment (jgeurten) [reply WDAC-Policy-Wizard/app/src/EventLogRuleConfiguration.cs]: Fixed all of these. Good feedback

File: WDAC-Policy-Wizard/app/src/EventLog.cs [hunk 3]
```diff
@@ -339,10 +352,22 @@ public static List<CiEvent> ReadPEBlockEvent(EventRecord entry, List<SignerEvent
...
+                        // If first/only signer, set the SignerInfo attribute to signer
+                        // otherwise, duplicate ciEvent and append to ciEvents
+                        if (ciEvent.SignerInfo.CorrelationId == null)
+                        {
+                            ciEvent.SignerInfo = signer;
+                            ciEvents.Add(ciEvent);
+                        }
+                        else
+                        {
+                            CiEvent ciEventCopy = ciEvent.Clone();
```
[the remark this answers is not in the document]
Comment (jgeurten) [reply WDAC-Policy-Wizard/app/src/EventLog.cs:367]: Correct. Calls MemberwiseClone

**Pull request merged**